modules/plugin/coverage-multidim/netcdf/src/test/java/org/geotools/coverage/io/netcdf/NetCDFMemoryMappedRandomAccessFileTest.java

```diff
@@ -0,0 +1,97 @@
+package org.geotools.coverage.io.netcdf;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import org.geotools.api.geometry.Position;
+import org.geotools.coverage.grid.GridCoverage2D;
+import org.geotools.coverage.grid.io.GridFormatFinder;
+import org.geotools.geometry.Position2D;
+import org.geotools.imageio.netcdf.utilities.NetCDFUtilities;
+import org.geotools.io.MemoryMappedRandomAccessFile;
+import org.geotools.referencing.crs.DefaultGeographicCRS;
+import org.geotools.test.TestData;
+import org.junit.Test;
+import ucar.nc2.dataset.NetcdfDataset;
+import ucar.nc2.iosp.AbstractIOServiceProvider;
+import ucar.unidata.io.RandomAccessFile;
+
+public class NetCDFMemoryMappedRandomAccessFileTest extends NetCDFBaseTest {
```

Comment:
The test is setting a memoryMapping limit.
Then there is a test that uses a file smaller than the memory limit (so that it can get fully mapped): testMemoryMapping
and a test that uses a file bigger than the memory limit (so that it needs to be mapped in chunks):
testMemoryMappingPartialBuffering

This test is also requesting sections that involve the mapping going forward and back.